registration/include/pcl/registration/impl/gicp.hpp

```diff
@@ -339,6 +338,33 @@ pcl::GeneralizedIterativeClosestPoint<PointSource, PointTarget>::OptimizationFun
   gicp_->computeRDerivative(x, R, g);
 }
 
+////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget> inline BFGSSpace::Status
+pcl::GeneralizedIterativeClosestPoint<PointSource, PointTarget>::OptimizationFunctorWithIndices::checkGradient(const Vector6d& g)
+{
+  auto linear_epsilon = gicp_->linear_gradient_tolerance_;
+  auto angular_epsilon = gicp_->angular_gradient_tolerance_;
+
+  if ((linear_epsilon < 0.) || (angular_epsilon < 0.))
+    return BFGSSpace::NegativeGradientEpsilon;
+
+  // express linear gradient as norm of translation parameters
+  auto linear_grad = g.head(3).norm();
```

Comment:
Should this be used somewhere - I can't seem to find if it is used?

Reply:
@larshg : I don't know if that's your question, but I think BFGS solver has only been added for GICP, and I don't think it is used anywhere else in PCL.

Reply:
No, it is more explicit the linear_grad variable.

Should it have been used in line 362 - or is it correct with the linear_epsilon < 0. ?

Reply:
@larshg : oh you're right, that's an obvious mistake, I'll fix this asap.

Reply:
fixed in 238a793